applications/tasks/src/kz_notify_resend.erl

```diff
@@ -0,0 +1,451 @@
+        {'error', _}=Error -> Error
+    end.
+
+%% @private
+-spec next() -> 'ok'.
+next() ->
+    gen_server:cast(?SERVER, 'next_cycle').
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%% Handling call messages
+%%
+%% @spec handle_call(Request, From, State) ->
+%%                                   {reply, Reply, State} |
+%%                                   {reply, Reply, State, Timeout} |
+%%                                   {noreply, State} |
+%%                                   {noreply, State, Timeout} |
+%%                                   {stop, Reason, Reply, State} |
+%%                                   {stop, Reason, State}
+%% @end
+%%--------------------------------------------------------------------
+-spec handle_call(any(), pid_ref(), state()) -> handle_call_ret_state(state()).
+handle_call('running', _From, #state{running=Running}=State) ->
+    {'reply', Running, State};
+handle_call(_Request, _From, State) ->
+    {'reply', {'error', 'not_implemented'}, State}.
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%% Handling cast messages
+%%
+%% @spec handle_cast(Msg, State) -> {noreply, State} |
+%%                                  {noreply, State, Timeout} |
+%%                                  {stop, Reason, State}
+%% @end
+%%--------------------------------------------------------------------
+-spec handle_cast(any(), state()) -> handle_cast_ret_state(state()).
+handle_cast('next_cycle', State) ->
+    {'noreply', State#state{running=[]}, ?TIME_BETWEEN_CYCLE};
+handle_cast(stop, State) ->
+    lager:debug("notify resender has been stopped"),
+    {stop, normal, State};
+handle_cast(_Msg, State) ->
+    lager:debug("unhandled cast: ~p", [_Msg]),
+    {'noreply', State}.
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%% Handling all non call/cast messages
+%%
+%% @spec handle_info(Info, State) -> {noreply, State} |
+%%                                   {noreply, State, Timeout} |
+%%                                   {stop, Reason, State}
+%% @end
+%%--------------------------------------------------------------------
+-spec handle_info(any(), state()) -> handle_info_ret_state(state()).
+handle_info('timeout', State) ->
+    ViewOptions = [{'startkey', 0}
+                  ,{'endkey', kz_time:current_tstamp()}
+                  ,{'limit', ?READ_LIMIT}
+                  ,'include_docs'
+                  ],
+    case kz_datamgr:get_results(?KZ_PENDING_NOTIFY_DB, <<"pending_notify/list_by_modified">>, ViewOptions) of
+        {'ok', []} -> {'noreply', State, ?TIME_BETWEEN_CYCLE};
+        {'ok', Pendings} ->
+            process_then_next_cycle([kz_json:get_value(<<"doc">>, J) || J <- Pendings]),
+            {'noreply', State#state{running=Pendings}};
+        {'error', 'not_found'} ->
+            kapps_maintenance:refresh(?KZ_PENDING_NOTIFY_DB),
```

Comment:
This case clause returns `{'noreply', State, ?TIME_BETWEEN_CYCLE};`.
The clause above after processing will call `next()` which initiate the timer.